lib/image-stream/archive.js

```diff
@@ -195,6 +195,7 @@ exports.extractImage = (archive, hooks) => {
     }).then((results) => {
       results.metadata.stream = results.imageStream;
       results.metadata.transform = new PassThroughStream();
+      results.metadata.path = archive;
```

Comment:
Could we add some unit-tests to check that when given e.g. `/path/to/rpi.img.xz` or `/path/to/rpi.zip` then the path stays as `/path/to/rpi.img.xz` or `/path/to/rpi.zip` and doesn't get transformed to `/path/to/rpi.img` or `rpi.img` ?